llvm/include/llvm/IR/IntrinsicsAMDGPU.td

```diff
@@ -391,6 +391,17 @@ def int_amdgcn_s_wait_loadcnt        : AMDGPUWaitIntrinsic;
 def int_amdgcn_s_wait_samplecnt      : AMDGPUWaitIntrinsic;
 def int_amdgcn_s_wait_storecnt       : AMDGPUWaitIntrinsic;
 
+// Force the VGPR allocation of the current wave to (at least) the given value.
```

Comment:
We should guarantee to not allocate above the requested value, except for the rounding part. Maybe instead say:

Request the hardware to allocate exactly the given number of VGPRs, rounded up to match hardware block boundaries.